src/web/WebSettingsService.cpp

```diff
@@ -88,90 +87,115 @@ void WebSettings::read(WebSettings & settings, JsonObject root) {
+            if (!System::is_valid_gpio(data[0], psram) || !System::is_valid_gpio(data[1], psram) || !System::is_valid_gpio(data[2], psram)
+                || !System::is_valid_gpio(data[3], psram) || !System::is_valid_gpio(data[4], psram) || !System::is_valid_gpio(data[6], psram)) {
+                settings.board_profile = "default"; // reset to factory default
+            }
+        } else {
+            settings.board_profile = "default"; // can't find profile, use "default"
         }
-    } else {
-        settings.board_profile = ""; // reset to factory default
     }
-    if (settings.board_profile == "") {
-        // unknown, check for NVS or scan for ethernet, use default E32/E32V2/S32
-        settings.board_profile = EMSESP::nvs_.getString("boot");
-        if (!System::load_board_profile(data, settings.board_profile.c_str())) {
+
+    // still don't have a valid board profile. Let's see if we can determine one
+    if (settings.board_profile == "default") {
+#ifdef EMSESP_DEBUG
+        EMSESP::logger().debug("Trying to detect board and set board profile...");
+#endif
+
 #if defined(EMSESP_STANDALONE)
-            settings.board_profile = "S32";
+        settings.board_profile = "S32";
 #elif CONFIG_IDF_TARGET_ESP32
-            if (settings.board_profile == "" && !psram) { // empty: new test for E32
+        // check for no PSRAM, could be a E32 or S32
+        if (!psram) {
 #if ESP_ARDUINO_VERSION_MAJOR < 3
-                if (ETH.begin(1, 16, 23, 18, ETH_PHY_LAN8720, ETH_CLOCK_GPIO0_IN)) {
+            if (ETH.begin(1, 16, 23, 18, ETH_PHY_LAN8720, ETH_CLOCK_GPIO0_IN)) {
 #else
-                if (ETH.begin(ETH_PHY_LAN8720, 1, 23, 18, 16, ETH_CLOCK_GPIO0_IN)) {
+            if (ETH.begin(ETH_PHY_LAN8720, 1, 23, 18, 16, ETH_CLOCK_GPIO0_IN)) {
 #endif
-                    EMSESP::nvs_.putString("boot", "E32");
-                } else {
-                    EMSESP::nvs_.putString("boot", "Test");
-                }
-            } else if (settings.board_profile == "Test" || psram) { // test E32V2
+                settings.board_profile = "E32";
+            } else {
+                settings.board_profile = "S32";
+            }
+        } else {
+// check for PSRAM, could be a E32V2 or S3
 #if ESP_ARDUINO_VERSION_MAJOR < 3
-                if (ETH.begin(0, 15, 23, 18, ETH_PHY_LAN8720, ETH_CLOCK_GPIO0_OUT)) {
+            if (ETH.begin(0, 15, 23, 18, ETH_PHY_LAN8720, ETH_CLOCK_GPIO0_OUT)) {
 #else
-                if (ETH.begin(ETH_PHY_LAN8720, 0, 23, 18, 15, ETH_CLOCK_GPIO0_OUT)) {
+            if (ETH.begin(ETH_PHY_LAN8720, 0, 23, 18, 15, ETH_CLOCK_GPIO0_OUT)) {
 #endif
-                    EMSESP::nvs_.putString("boot", "E32V2");
-                } else {
-                    EMSESP::nvs_.putString("boot", "S32");
-                }
+                settings.board_profile = "E32V2";
             } else {
-                EMSESP::nvs_.putString("boot", "S32");
+                settings.board_profile = "S3";
```

Comment:
This is section TARGET_ESP32, t could never be a S3-chip (what is named `S32S3`), a S3 board profile does not exist. This should also default to S32 or CUSTOM with S32 identical gpios.